### DISCUSSION TITLE
Is Threading.Analyzers installed by default?

### DISCUSSION THREAD

**Discussion by maxkoshevoi** [microsoft/vs-threading#926] [Q&A] [answered]
Are rules from `Microsoft.VisualStudio.Threading.Analyzers` used by Visual Studio in all projects by default, or does it need to be installed explicetly?

**Comment by AArnott** [comment, marked as answer]
They are *not* included in projects by default. 
But they are transitively brought in by some packages. For example if your project references the VS SDK via nuget packages, you'll likely get these analyzers automatically. A few other non-VS libraries also may bring them in.

**Reply by maxkoshevoi** [reply]
What do you mean by `VS SDK`? Something used when developing VS extensions? Judging by the name, I thought, VS runs those analyzers on all open project. Is this package meant to be installed manually for projects that just use `Microsoft.NET.Sdk`? Again, name is what's confusing me.

**Reply by AArnott** [reply]
By "VS SDK", yes I mean the SDK referenced by projects that build extensions for the VS IDE.

The vs-threading analyzers include rules that may benefit any .NET project, whether they are VS-specific or not. But .NET projects in general do not include them. Some of these rules are being copied over into the .NET SDK actually, because they are so generally useful. But in the meantime, I often add a package reference to these analyzers in my own projects so I get the full benefit.
